packages/feed-list/src/index.ts

```diff
@@ -0,0 +1,226 @@
+import { Readable } from 'stream'
+import {
+  FEED_MAX_DATA_LENGTH,
+  BaseBzz,
+  BaseResponse,
+  FeedID,
+  FeedParams,
+  FetchOptions,
+} from '@erebos/api-bzz-base'
+import { createHex, Hex, hexInput } from '@erebos/hex'
+
+export const MAX_CHUNK_BYTE_LENGTH = FEED_MAX_DATA_LENGTH
+// Hex value is a hex string (= byte length x 2) prefixed by `0x`
+export const MAX_CHUNK_VALUE_LENGTH = MAX_CHUNK_BYTE_LENGTH * 2 + 2
+
+export interface ForwardsChunkIterator<T> extends AsyncIterator<T> {
+  length: number
+}
+
+export interface ChunkListReaderConfig<
+  Bzz extends BaseBzz<BaseResponse, Readable> = BaseBzz<BaseResponse, Readable>
+> {
+  bzz: Bzz
+  feed: FeedID | FeedParams
+  fetchOptions?: FetchOptions
+}
+
+export class ChunkListReader<
+  Bzz extends BaseBzz<BaseResponse, Readable> = BaseBzz<BaseResponse, Readable>
+> {
+  public bzz: Bzz
+  public fetchOptions: FetchOptions
+  protected id: FeedID
+
+  constructor(config: ChunkListReaderConfig<Bzz>) {
+    this.bzz = config.bzz
+    this.fetchOptions = config.fetchOptions || {}
+    this.id =
+      config.feed instanceof FeedID ? config.feed : new FeedID(config.feed)
+  }
+
+  public async load(index: number): Promise<Hex | null> {
+    const id = this.id.clone()
+    id.time = index
+
+    try {
+      const data = await this.bzz.getRawFeedChunkData(id, this.fetchOptions)
+      return createHex(Buffer.from(data))
+    } catch (err) {
+      if (err.status === 404) {
+        return null
+      }
+      throw err
+    }
+  }
+
+  public createBackwardsIterator(
+    initialIndex: number,
+  ): AsyncIterator<Hex | null> {
+    const id = this.id.clone()
+    id.time = initialIndex
+    return {
+      // @ts-ignore
+      [Symbol.asyncIterator]() {
+        return this
+      },
+      next: async (): Promise<IteratorResult<Hex | null>> => {
+        if (id.time === -1) {
+          return { done: true, value: undefined }
+        }
+        const value = await this.load(id.time)
+        // eslint-disable-next-line require-atomic-updates
+        id.time -= 1
+        return { done: false, value }
+      },
+    }
+  }
+
+  public createForwardsIterator(initialIndex = 0): ForwardsChunkIterator<Hex> {
+    const id = this.id.clone()
+    id.time = initialIndex
+    return {
+      // @ts-ignore
+      [Symbol.asyncIterator]() {
+        return this
+      },
+      get length(): number {
+        return id.time
+      },
+      next: async (): Promise<IteratorResult<Hex>> => {
+        const value = await this.load(id.time)
+        if (value === null) {
+          return { done: true, value: undefined }
+        }
+        // eslint-disable-next-line require-atomic-updates
+        id.time += 1
```

Comment:
I know it's kinda nitpicking at this stage, but this should throw an exception when it reaches the maximum value of `time` (which is 2^56-1).

Reply:
[Number.MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) is `2^53 - 1` so we might as well use this, no?
It's already a very high number for this kind of use case I think.

Reply:
Yeah, that's right